service/src/main/java/org/ehrbase/service/DirectoryServiceImp.java

```diff
@@ -163,6 +166,20 @@ public Folder update(UUID ehrId, Folder folder, ObjectVersionId ifMatches) {
 
     @Override
     public Folder update(UUID ehrId, Folder folder, ObjectVersionId ifMatches, UUID contributionId, UUID auditId) {
+
+        UUID uuid = UUID.fromString(ifMatches.getObjectId().getValue());
+
+        // Ensure UIDs matching
+        String folderUidValue = Optional.ofNullable(folder.getUid())
```

Comment:
update without an uid on Folder is fine because the id it taken from the If-Match header.  For the uid check I've added `hasFolderForVoId(UUID)` to the repository that is now used insted of the ehr & folder idx check.

Reply:
OK, but we need to make sure that the vo_id belongs to the ehr

Reply:
Changed check from `hasFolderForVoId(UID voId)` to `hasFolderInEhrForVoId(UID ehrId, UID voId)`.